lib/uplink/uplink.go

```diff
@@ -68,6 +78,11 @@ func (c *Config) setDefaults(ctx context.Context) error {
 	if c.Volatile.MaxInlineSize.Int() == 0 {
 		c.Volatile.MaxInlineSize = 4 * memory.KiB
 	}
+	if c.Volatile.MaxMem.Int() == 0 {
+		c.Volatile.MaxMem = 4 * memory.MiB
```

Comment:
maybe we should check `if c.Volatile.MaxMem.Int() <= 0`

Reply:
the intended behavior, documented up on lines 60-65, is that if you don't set this field, we get a zero value and we want to treat that as "use the library default". but we do still want to make it possible for the user to select the "use as little memory as possible" option supported by `ecclient.NewClient`. we can't use the zero value for that option because zero is the default and the default has to be to do the most reasonable thing (library default).

that's why we set MaxMem to 0 if the user passes a negative value (that's what `ecclient.NewClient` accepts if you want to tell it to allocate as little memory as possible) and we set it to 4MiB if the user passes the zero value.